src/GraphQL.Tests/Extensions/TypeExtensionTests.cs

```diff
@@ -10,66 +10,75 @@ public class TypeExtensionTests
+    [InlineData(60, true, false, typeof(Class1), "Class1")]
+    [InlineData(61, true, false, typeof(Class1Type), "Class1")]
+    [InlineData(62, true, false, typeof(Class1Graph), "Class1Graph")]
+    [InlineData(63, true, false, typeof(Class1GraphType), "Class1")]
+    [InlineData(64, true, false, typeof(GenericClass<>), "GenericClass")]
+    [InlineData(65, true, false, typeof(GenericClassType<>), "GenericClass")]
+    [InlineData(66, true, false, typeof(GenericClassGraph<>), "GenericClassGraph")]
+    [InlineData(67, true, false, typeof(GenericClassGraphType<>), "GenericClass")]
+    [InlineData(68, true, false, typeof(GenericClass<Class1>), "GenericClass")]
+    [InlineData(69, true, false, typeof(GenericClassType<Class1>), "GenericClass")]
+    [InlineData(70, true, false, typeof(GenericClassGraph<Class1>), "GenericClassGraph")]
+    [InlineData(71, true, false, typeof(GenericClassGraphType<Class1>), "GenericClass")]
+    [InlineData(72, true, false, typeof(GenericClass<Class1Type>), "GenericClass")]
+    [InlineData(73, true, false, typeof(GenericClass<Class1Graph>), "GenericClass")]
+    [InlineData(74, true, false, typeof(GenericClass<Class1GraphType>), "GenericClass")]
+    [InlineData(75, true, false, typeof(GenericClass<Class1, string>), "GenericClass")]
+    [InlineData(76, true, false, typeof(GenericClassType<Class1, string>), "GenericClass")]
+    [InlineData(77, true, false, typeof(GenericClassGraph<Class1, string>), "GenericClassGraph")]
+    [InlineData(78, true, false, typeof(GenericClassGraphType<Class1, string>), "GenericClass")]
+    [InlineData(79, true, true, typeof(Class1), "Class1")]
+    [InlineData(80, true, true, typeof(GenericClass<Class1>), "GenericClass")]
+    [InlineData(81, true, false, typeof(NonNullGraphType<StringGraphType>), "String")]
+    [InlineData(82, true, false, typeof(ListGraphType<StringGraphType>), "String")]
+    [InlineData(83, true, false, typeof(NonNullGraphType<ListGraphType<NonNullGraphType<StringGraphType>>>), "String")]
+    [InlineData(84, true, false, typeof(NonNullGraphType<ListGraphType<NonNullGraphType<InputObjectGraphType<Class1>>>>), "InputObject")]
+    [InlineData(85, true, false, typeof(ConnectionType<StringGraphType>), "Connection")]
+    [InlineData(86, true, false, typeof(Connection<string>), "Connection")]
+    [InlineData(87, true, false, typeof(Test1<string>.Test2), "Test2")]
+    [InlineData(88, true, false, typeof(Test1<string>.Test3<Class1>), "Test3")]
+    [InlineData(89, true, true, typeof(Test1<string>.Test2), "Test2")]
+    [InlineData(90, true, true, typeof(Test1<string>.Test3<Class1>), "Test3")]
 
     // note: cannot test F# anonymous class names within C# tests
-    public void GraphQLNameTest(bool useLegacyTypeNaming, bool useDeclaringTypeNames, Type type, string expected)
+    public void GraphQLNameTest(int i, bool useLegacyTypeNaming, bool useDeclaringTypeNames, Type type, string expected)
```

Comment:
Added numbering.  Added tests 37-40 and 87-90.  Tests 37, 39, 87, and 89 failed because of bug 1 before.  Tests 38 and 40 failed because of bug 2 before.